src/test/java/javaslang/collection/AbstractMapTest.java

```diff
@@ -304,6 +302,11 @@ public void shouldReturnTuple2SetOfANonEmptyMap() {
         assertThat(emptyMap().put(1, "1").put(2, "2").toSet()).isEqualTo(HashSet.ofAll(Tuple.of(1, "1"), Tuple.of(2, "2")));
     }
 
+    @Test
+    public void shouldReturnModifiedValuesMap() {
+        assertThat(HashMap.of(1, "1").put(2, "2").mapValues(Integer::parseInt)).isEqualTo(HashMap.of(1, 1).put(2, 2));
```

Comment:
`stupid mistakes` do not exist :-) It is the 'normal' dev workflow - hacking - reviewing - fixing - ...

It is fun!